rocket/src/main/scala/ipxact/IpxactGeneratorTransform.scala

```diff
@@ -0,0 +1,419 @@
+// See LICENSE for license details.
+
+package ipxact
+
+import java.io.{File, PrintWriter}
+
+import firrtl.annotations._
+import firrtl.ir._
+import firrtl.{
+  CircuitForm, CircuitState, Driver, ExecutionOptionsManager, HasFirrtlOptions, HighForm, LowForm, Parser, Transform, _
+}
+import freechips.rocketchip.diplomacy.{AddressMapEntry, LazyModule}
+import freechips.rocketchip.util.{AddressMapAnnotation, ParamsAnnotation, RegFieldDescMappingAnnotation}
+import ipxact.IpxactGeneratorTransform.{indent, lineWidth}
+import ipxact.devices.AXI4Device
+
+/**
+  * This transform builds a xml object from the annotations
+  * and firrtl circuit
+  */
+class IpxactGeneratorTransform extends Transform {
+  override def inputForm: CircuitForm = LowForm
+
+  override def outputForm: CircuitForm = LowForm
+
+  /**
+    * Adds parameters to the parameters component
+    * @param annotation the firrtl annotation for parameters
+    * @return
+    */
+  def generateParameter(annotation: ParamsAnnotation): Seq[scala.xml.Node] = {
+    val prefix = annotation.paramsClassName
+    annotation.params.map { case (paramName, value) =>
+      <spirit:parameter>
```

Comment:
Man, this DSL is cool

Reply:
Dotty will be dropping this in favor of an xml"" interpolator, we'll cross that bridge when we come to it.